### PR TITLE
Display: Added a option to turn off the raspberry pi through the display

### PR DESCRIPTION
**Pull request by paulodrugos**
Signed-off-by: Paulo Drugos paulodrugos@gmail.com

### CODE REVIEW

**Comment by blaskkaffe** [comment]
Nice! Have been looking for this!

**Comment by KevinOConnor** [comment]
Thanks.  I think this change would need to be added to a new "extras" module instead of changing gcode.py - as gcode.py is intended only for "core" g-code commands.

-Kevin

File: klippy/gcode.py [hunk 1]
```diff
@@ -710,3 +710,8 @@ def cmd_HELP(self, params):
             if cmd in self.gcode_help:
                 cmdhelp.append("%-10s: %s" % (cmd, self.gcode_help[cmd]))
         self.respond_info("\n".join(cmdhelp))
+    cmd_RPI_SHUTDOWN_when_not_ready = True
+    cmd_RPI_SHUTDOWN_help = "Turn octoprint and the raspberry off"
+    def cmd_RPI_SHUTDOWN(self, params):
+        os.system("sudo shutdown -h now")
```
Comment (trevjonez) [inline klippy/gcode.py]: this also assumes sudo access and that all of these params are available. fairly certain klipper can be ran on more than raspian so you would likely want to ensure the commands are going to work before exposing the menu option or gcode? at the least a try catch and error message?

**Comment by paulodrugos** [comment]
Ok.

I will move this to a module and ensure the correct execution of the shutdown command when I have free time...

**Pull request closed without merging**